website/docs/topics/gpt-assistant/gpt_assistant.md

```diff
@@ -0,0 +1,151 @@
+
+2. Code-Based Uploading: Alternatively, you can upload files and retrieve their file IDs programmatically using the following code snippet:
+
+    ```python
+    from openai import OpenAI
+    client = OpenAI(
+        # Defaults to os.environ.get("OPENAI_API_KEY")
+    )
+    # Upload a file with an "assistants" purpose
+    file = client.files.create(
+      file=open("mydata.csv", "rb"),
+      purpose='assistants'
+    )
+    ```
+
+### File Search
+
+The [File Search](https://platform.openai.com/docs/assistants/tools/file-search) tool empowers your agents to tap into knowledge beyond its pre-trained model. This allows you to incorporate your own documents and data, such as product information or code files, into your agent's capabilities.
+
+Using the File Search with the following configuration.
+
+```python
+assistant_config = {
+    "tools": [
+        {"type": "file_search"},
+    ],
+    "tool_resources": {
+        "file_search": {
+            "vector_store_ids": ["$vector_store.id"]
+        }
+    }
+}
+```
+
+Here's how to obtain the vector_store.id using two methods:
+
+1. OpenAI Playground: Leverage the OpenAI Playground, an interactive platform accessible at https://platform.openai.com/playground, to create a vector store, upload your files, and add it into your vector store. Once complete, you'll be able to retrieve the associated `vector_store.id`.
+
+2. Code-Based Uploading:Alternatively, you can upload files and retrieve their file IDs programmatically using the following code snippet:
+
+    ```python
+    from openai import OpenAI
+    client = OpenAI(
+        # Defaults to os.environ.get("OPENAI_API_KEY")
+    )
+
+    # Step 1: Create a Vector Store
+    vector_store = client.beta.vector_stores.create(name="Financial Statements")
+    print("Vector Store created:", vector_store.id)  # This is your vector_store.id
+
+    # Step 2: Prepare Files for Upload
+    file_paths = ["edgar/goog-10k.pdf", "edgar/brka-10k.txt"]
+    file_streams = [open(path, "rb") for path in file_paths]
+
+    # Step 3: Upload Files and Add to Vector Store (with status polling)
+    file_batch = client.beta.vector_stores.file_batches.upload_and_poll(
+        vector_store_id=vector_store.id, files=file_streams
+    )
+
+    # Step 4: Verify Completion (Optional)
+    print("File batch status:", file_batch.status)
+    print("Uploaded file count:", file_batch.file_counts.processed)
+    ```
+
+### Function calling
+
+Function Calling empowers you to extend the capabilities of your agents with your pre-defined functionalities, which allows you to describe custom functions to the Assistant, enabling intelligent function selection and argument generation.
+
+Using the Function calling with the following configuration.
+
+```python
+# learn more from https://platform.openai.com/docs/guides/function-calling/function-calling
+api_schema = "define your function schema here"
```

Comment:
Here, we can create an actual function e.g., `get_current_weather`, and use the `get_function_schema` to generate the JSON schema. 

https://github.com/microsoft/autogen/blob/e3ccf228e23940f7e27c28ea71dc3167fd6cb6e1/autogen/function_utils.py#L211-L212